### PR TITLE
`TASK115` Move action to its own repository

### PR DESCRIPTION
**Pull request by aoozdemir**

### CODE REVIEW

File: src/instrument.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import * as core from '@actions/core'
+import * as exec from '@actions/exec'
+import * as tc from '@actions/tool-cache'
+import { resolve } from 'path'
+
+const THUNDRA_AGENT_REPOSITORY =
+    'https://thundra-release-lab.s3-us-west-2.amazonaws.com/thundra-agent/thundra-agent-bootstrap.jar'
+
+const MAVEN_INSTRUMENTATION_SCRIPT =
```
Comment (serkan-ozal) [inline src/instrument.ts:9]: Can you get the latest version?

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name: "Thundra Maven Test Instrumentation"
+description: "Thundra Maven Test Instrumentation"
+author: "Oguzhan Ozdemir <oguzhan.ozdemir@thundra.io>"
+inputs:
```
Comment (serkan-ozal) [inline action.yml:4]: Can we define optional `instrumenter_version` input to be able to specify maven instrumenter version.

Comment (serkan-ozal) [inline action.yml:4]: Can we define optional `agent_version` input to be able to specify agent version.

File: action.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  apikey:
+    required: true
+    description: "Thundra's API Key"
+  project_id:
+    required: false
+    description: "Thundra's Test Project ID"
+  command:
+    required: true
+    description: "Command to run your tests"
+  github_token:
```
Comment (serkan-ozal) [inline action.yml:14]: I think we can get rid of `github_token` input for now

File: action.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    required: false
+    description: "Github Token to comment a message to the linked PR"
+  message:
```
Comment (serkan-ozal) [inline action.yml:17]: I think we can get rid of `message` input for now

File: src/main.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        core.startGroup('[Thundra] Execute the given command...')
+        core.info(`> Instrumenting the application`)
+        await instrument(command)
+        core.endGroup()
+
+        core.info(`[Thundra] Executing the command`)
+        await exec.exec(`sh -c "${command}"`)
+
+        // Comment to PR
+        if (github.context.payload.pull_request && github_token && message) {
```
Comment (serkan-ozal) [inline src/main.ts:31]: I think, we can get rid of this part as probably it will be provided by a Github bot

File: src/main.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
+import * as core from '@actions/core'
+import * as exec from '@actions/exec'
+import * as github from '@actions/github'
+
+import { instrument } from './instrument'
+
+const apikey: string = core.getInput('apikey')
```
Comment (serkan-ozal) [inline src/main.ts:7]:
```js
const apikey: string = core.getInput('apikey', {required: true})
```

File: src/main.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+const project_id: string = core.getInput('project_id')
+const command: string = core.getInput('command')
+const github_token: string = core.getInput('github_token')
```
Comment (serkan-ozal) [inline src/main.ts:10]: remove

File: src/main.ts [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+const message: string = core.getInput('message')
```
Comment (serkan-ozal) [inline src/main.ts:11]: remove

File: src/version.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+export async function getVersion(url: string, version?: string): Promise<string | undefined> {
+    const response = await fetch(url)
+    const json = await xml2json(await response.text())
+
+    const availableVersions: string[] = json.metadata.versioning[0].versions[0].version
+    const latestVersion: string = json.metadata.versioning[0].release[0]
+
+    if (version && availableVersions.find(v => v === version)) {
+        return availableVersions.find(v => v === version)
```
Comment (serkan-ozal) [inline src/version.ts:12]: just return `version`. Do we need to `availableVersions.find(v => v === version)` here again?

Comment (aoozdemir) [reply src/version.ts:12]: Yep, there is no need. I simply missed this obvious thing. Sorry :)

**Review by serkan-ozal — APPROVED** [review]

**Review by tolgatakir — APPROVED** [review]

**Review by serkan-ozal — APPROVED** [review]

**Pull request merged**